source/core/DualTypeVisitor.cpp

```diff
@@ -129,23 +129,13 @@ namespace Vireo
         Boolean isTypeXIntrinsicClusterType = typeRefX->IsIntrinsicClusterDataType(&typeXName);
         Boolean isTypeYIntrinsicClusterType = typeRefY->IsIntrinsicClusterDataType(&typeYName);
         if (isTypeXIntrinsicClusterType && isTypeYIntrinsicClusterType) {
-            success = IntrinsicClustersCompatible(typeRefX, pDataX, typeRefY, pDataY, operation);
+            success = operation->AreIntrinsicClustersCompatible(typeRefX, typeRefY);
```

Comment:
As operation cannot be null (it is always one of the derivative objects of DualTypeOperation, it will be cleaner to mark the parameter as "const DualTypeOperation &" throughout in all the APIs. That will in turn make the methods of DualTypeOperation const -- which is true today but just not specified explicitly. We should just fix it.

Reply:
Done.